pkg/cloudprovider/instance.go

```diff
@@ -140,21 +139,13 @@ func (p *InstanceProvider) GetByID(ctx context.Context, id string) (*ec2.Instanc
 	return instances[0], nil
 }
 
-func (p *InstanceProvider) Get(ctx context.Context, machineName string) (*ec2.Instance, error) {
-	instances, err := p.List(ctx, machineName)
-	if err != nil {
-		return nil, err
-	}
-	return instances[0], nil
-}
-
-func (p *InstanceProvider) List(ctx context.Context, machineName string) ([]*ec2.Instance, error) {
+func (p *InstanceProvider) List(ctx context.Context) ([]*ec2.Instance, error) {
 	// Use the machine name data to determine which instances match this machine
 	out, err := p.ec2api.DescribeInstancesWithContext(ctx, &ec2.DescribeInstancesInput{
 		Filters: []*ec2.Filter{
 			{
-				Name:   aws.String(fmt.Sprintf("tag:%s", v1alpha5.MachineNameLabelKey)),
-				Values: aws.StringSlice([]string{machineName}),
+				Name:   aws.String(fmt.Sprintf("tag:%s", v1alpha5.ProvisionerNameLabelKey)),
+				Values: aws.StringSlice([]string{"*"}),
```

Comment:
use the tag-key filter instead:

```suggestion
				Name:   aws.String("tag-key"),
				Values: aws.StringSlice([]string{v1alpha5.ProvisionerNameLabelKey}),
```